core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -3570,6 +3591,72 @@ public long generateSeqNo() {
         }
     }
 
+    public void testMinGenerationForSeqNo() throws IOException, BrokenBarrierException, InterruptedException {
```

Comment:
I wonder if this test tests what we should test on the engine level. We already test on the translog level that  `translog.getMinGenerationForSeqNo()` works as designed. I think that on the engine level we should make sure that recovery doesn't lose ops and recover exactly what we want. We can also maybe check that the commit point has the meta data we expect.  wdyt?